lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -62,9 +65,47 @@ sub _save_vars($) {
     close($fd);
 }
 
+sub cache_assets {
+    my ($vars) = @_;
+    #cache ISO and HDD
+    for my $this_asset (sort grep { /^(ISO|HDD)[_]*\d?$/ } keys %$vars) {
+        log_debug("Found $this_asset, caching " . $vars->{$this_asset});
+        my $asset = get_asset($job, $this_asset, $vars->{$this_asset});
+        return {error => "Can't download $vars->{$this_asset}"} unless $asset;
+        symlink($asset, basename($asset)) or die "cannot create link: $asset, $pooldir";
+    }
+    return undef;
+}
+
+sub cache_tests {
+
+    my ($shared_cache, $testpoolserver) = @_;
+
+    my $start = time;
+    #Do an flock to ensure only one worker is trying to synchronize at a time.
+    my @cmd = qw(flock -E 999);
```

Comment:
so that would serialize the syncing job of all workers and hinder scalability, right? Why is this necessary? The worker should access the server r/o anyway so for reasons to serialize the I/O on the server?